File: iphone/MyFirstApp/MyFirstApp/ViewController.m
//
//  ViewController.m
//  MyFirstApp
//
//  Created by taisuke fujita on 12/05/27.
//  Copyright (c) 2012年 Revolution 9. All rights reserved.
//

#import "ViewController.h"
#import "ModalViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad
{
    [super viewDidLoad];
    /*
    UIView *view = [[UIView alloc] initWithFrame:CGRectMake(0, 0, 100, 100)];
    view.backgroundColor = [UIColor redColor];
    [self.view addSubview:view];
     */
    UIButton *button = [UIButton buttonWithType:UIButtonTypeRoundedRect];
    button.frame = CGRectMake(0, 0, 100, 44);
    [button setTitle:@"button" forState:UIControlStateNormal];
    [button addTarget:self action:@selector(respondToButtonClick:) forControlEvents:UIControlEventTouchUpInside];
    [self.view addSubview:button];
}

- (void)viewDidUnload
{
    [super viewDidUnload];
    // Release any retained subviews of the main view.
}

- (BOOL)shouldAutorotateToInterfaceOrientation:(UIInterfaceOrientation)interfaceOrientation
{
    if ([[UIDevice currentDevice] userInterfaceIdiom] == UIUserInterfaceIdiomPhone) {
        return (interfaceOrientation != UIInterfaceOrientationPortraitUpsideDown);
    } else {
        return YES;
    }
}

- (void)respondToButtonClick:(id)sender
{
    ModalViewController *controller = [[ModalViewController alloc] init];
    controller.view.backgroundColor = [UIColor whiteColor];
    [self presentViewController:controller animated:YES completion:nil];
}

@end

File: ios_swift/TransitionSample/TransitionSample/ViewController.swift
//
//  ViewController.swift
//  TransitionSample
//
//  Created by taisuke fujita on 2014/12/05.
//  Copyright (c) 2014年 taisuke fujita. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        
        let button = UIButton.buttonWithType(UIButtonType.System)
            as UIButton
        button.frame = CGRect(x: 0, y: 0, width: 50, height: 44)
        button.setTitle("開く", forState: UIControlState.Normal)
        button.addTarget(self,
            action: "respondToButtonClick:",
            forControlEvents: UIControlEvents.TouchUpInside)
        view.addSubview(button);
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    func respondToButtonClick(sender: UIButton!) {
        let controller = ModalViewController()
        controller.view.backgroundColor = UIColor.whiteColor()
        self.presentViewController(controller, animated: true, completion: nil)
    }

}

